ribbon-loadbalancer/src/main/java/com/netflix/client/PrimeConnections.java

```diff
@@ -357,20 +357,19 @@ private Boolean connectToServer(final Server s, final PrimeConnectionListener li
                 lastException = e;
                 sleepBeforeRetry(tryNum);
             } 
-            logger.debug("server:" + s + ", result=" + success + ", tryNum="
-                    + tryNum + ", maxRetries=" + maxRetries);
+            logger.debug("server:{}, result={}, tryNum={}, maxRetries={}", s, success, tryNum, maxRetries);
             tryNum++;
         } while (!success && (tryNum <= maxRetries));
         // set the alive flag so that it can be used by load balancers
         if (listener != null) {
             try {
                 listener.primeCompleted(s, lastException);
-            } catch (Throwable e) {
-                logger.error("Error calling PrimeComplete listener", e);
+            } catch (Exception e) {
+                logger.error("Error calling PrimeComplete listener for server '{}'", s.getHost(), e);
```

Comment:
It looks like this could be the first dereference of "s" in this function, if it could be null, please null check it before logging data from it.